src/test/java/com/open/boot/core/utils/CommonUtilTest.java

```diff
@@ -0,0 +1,107 @@
+package com.open.boot.core.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+
+public class CommonUtilTest {
+
+    @Test
+    public void testIsEmpty() {
+        Assert.assertTrue(CommonUtil.isEmpty(""));
+        Assert.assertTrue(CommonUtil.isEmpty(null));
+
+        Assert.assertFalse(CommonUtil.isEmpty("foo"));
+    }
+
+    @Test
+    public void testGetStringSimple() {
+        Assert.assertEquals("foob***obar",
+                CommonUtil.getStringSimple("foobar"));
+    }
+
+    @Test
+    public void testGetYestoday() {
+        Date date =
+                new Date(System.currentTimeMillis() - 24 * 60 * 60 * 1000);
+        String data = new SimpleDateFormat("yyyyMMdd").format(date);
+        Assert.assertEquals(data, CommonUtil.getYestoday());
+    }
+
+    @Test
+    public void testGetCurrentDate() {
+        Date date = new Date();
+        String data = new SimpleDateFormat("yyyyMMdd").format(date);
+        Assert.assertEquals(data, CommonUtil.getCurrentDate());
+    }
+
+    @Test
+    public void testGetCurrentTime() {
+        Date date = new Date();
+        String data = new SimpleDateFormat("yyyyMMddHHmmss").format(date);
+        Assert.assertEquals(data, CommonUtil.getCurrentTime());
+    }
+
+    @Test
+    public void testGetGenderByIdNumber() {
+        Assert.assertEquals(1,
+                (int) CommonUtil.getGenderByIdNumber("123456789012345"));
+        Assert.assertEquals(1,
+                (int) CommonUtil.getGenderByIdNumber("123456789012345678"));
+    }
+
+    @Test
+    public void testGetBirthdayByIdNumber() {
+        Assert.assertEquals("1978-90-12",
+                CommonUtil.getBirthdayByIdNumber("123456789012345"));
+        Assert.assertEquals("7890-12-34",
+                CommonUtil.getBirthdayByIdNumber("123456789012345678"));
+    }
+
+    @Test
+    public void testGenRandomNum() {
+        Assert.assertEquals("", CommonUtil.genRandomNum(0));
```

Comment:
Add a couple of tests with `pwd_len` >0, only assert that the length of the string is equal to `pwd_len`

Reply:
Done